src/main/java/com/algorithmia/algorithmHandler/Response.java

```diff
@@ -0,0 +1,46 @@
+package com.algorithmia.algorithmHandler;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import org.apache.commons.codec.binary.Base64;
+
+class Response<T> {
+    private MetaData metaData;
+    private T result;
+    public class MetaData{
+        private String content_type;
+        MetaData(String contentType){this.content_type = contentType;}
+    }
+    Response(Object rawData){
+        String contentType;
+        T data;
+        if(rawData == null){
+            contentType = "json";
+            data = null;
+        }
+        else if(rawData instanceof String){
+            contentType = "text";
+            data = (T)rawData;
+        }
+        else if (rawData instanceof byte[]){
+            contentType = "binary";
+            data = (T)Base64.encodeBase64String((byte[])rawData);
+
+        } else {
+            contentType = "json";
+            data = (T)rawData;
+        }
+
+        metaData = new MetaData(contentType);
+        result = data;
+    }
+    String getJsonOutput(){
+        Gson gson = new Gson();
```

Comment:
Gson is not stateful and thread-safe; you should make it a class variable rather than a method variable.

Reply:
Agreed, I'll refactor this.